modules/protocol/src/vector.ts

```diff
@@ -332,6 +393,97 @@ export class Vector implements IVectorProtocol {
     return undefined;
   }
 
+  // Adds a given task to the internal queue
+  // TODO: implement
+  private addToQueue(
+    params: UpdateParams<any>,
+    previous?: ChannelUpdate,
+  ): Promise<
+    Result<
+      {
+        updatedChannel: FullChannelState;
+        updatedActiveTransfers?: FullTransferState[];
+        updatedTransfer?: FullTransferState;
+      },
+      QueuedUpdateError
+    >
+  > {
+    throw new Error("addToQueue method not implemented");
+  }
+
+  // Adds a given task to the front of the internal queue
+  // TODO: implement
+  private addToFrontOfQueue(
+    params: UpdateParams<any>,
+    previous?: ChannelUpdate,
+  ): Promise<
+    Result<
+      {
+        updatedChannel: FullChannelState;
+        updatedActiveTransfers?: FullTransferState[];
+        updatedTransfer?: FullTransferState;
+      },
+      QueuedUpdateError
+    >
+  > {
+    throw new Error("addToQueue method not implemented");
+  }
+
+  /**
+   * Returns true if the received upddate includes our proposed update. If true,
+   * this means you don't have to re-send your proposed update. If false, this
+   * means you should resend your proposed upddate
+   * @param receivedUpdate The update you have gotten from your counterparty
+   * @param proposedUpdate The update you have sent out to counterparty
+   */
+  private includesOurProposedUpdate(
+    receivedUpdate: ChannelUpdate,
+    proposedUpdate: ChannelUpdate,
+    // TODO: should probably just use the generated merkle tree here instead
+    // of regenerating it from the transfers and channel
+    updatedActiveTransfers: FullTransferState[],
+    proposedTransfer?: FullTransferState,
+  ): Result<boolean, QueuedUpdateError> {
+    // If both are a setup update, your update would fail and the ultimate
+    // result (a channel is set up) is achieved
+    if (receivedUpdate.type === UpdateType.setup && proposedUpdate.type === UpdateType.setup) {
```

Comment:
After some further consideration I think this specific case to be invalid. Both participants cannot be Alice in the contract that was actually deployed on-chain. Only Alice should send update 0.

That leaves only the "try two" method for all cases, where if the wrong participant performs setup it should fail for that validation.